### PR TITLE
Upgrade the APIs version to support kubernetes 1.25

### LINKED ISSUE #928: policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
Starting the operator shows the log in the title.
Version 2.8.0

### PR DESCRIPTION
**Pull request by prafull01**
Fixes #928 

To support public operator on k8s 1.25+, We have to upgrade following underlying dependencies.

- Kubernetes and client go from 1.20.2 to 1.21.2
- Controller-runtime from 0.8.2 to 0.9.2

Screenshot 2023-01-11 at 4 32 27 PM

### CODE REVIEW

**Review by pseudomuto — APPROVED** [review]
This all looks legit to me 👍 Thanks for doing this.

Is there anything we need to do to not break people's clusters if they upgrade?

File: WORKSPACE [hunk 1]
```diff
@@ -108,19 +108,15 @@ container_pull(
...
+    sha256 = "ce5b9bc0926681e2e7f2147b49096f143e6cbc783e71bc1d4f36ca76b00e6f4a",
+    strip_prefix = "rules_k8s-0.7",
+    urls = ["https://github.com/bazelbuild/rules_k8s/archive/refs/tags/v0.7.tar.gz"],
 )
 
 load("@io_bazel_rules_k8s//k8s:k8s.bzl", "k8s_repositories")
 
 k8s_repositories()
 
-load("@io_bazel_rules_k8s//k8s:k8s_go_deps.bzl", k8s_go_deps = "deps")
```
[the remark this answers is not in the document]
Comment (prafull01) [reply WORKSPACE:120]: Yes once we have updated the Bazel rules for k8s, these calls were not needed. This was like loading the rules again and with this call Bazel was throwing error. I have removed this call as per Bazel recommendation on k8s rules.

**Comment by prafull01** [comment]
People's cluster are safe. Everything remains same just the PDB moved from policy/v1beta1 to policy/v1 which doesn't have any impact on the cluster itself.

**Comment by pseudomuto** [comment]
@prafull01 mind adding an entry to CHANGELOG.md?

**Review by prafull01 — COMMENTED** [review]
@pseudomuto Added Change log 👍

**Pull request merged**